src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSDeleteOnRestartManager.cs

```diff
@@ -167,16 +171,13 @@ public void CheckAndRaisePackageDirectoriesMarkedForDeletion()
             "Microsoft.Design",
             "CA1031:DoNotCatchGeneralExceptionTypes",
             Justification = "We want to log an exception as a warning and move on")]
-        public void DeleteMarkedPackageDirectories(INuGetProjectContext projectContext)
+        public async Task DeleteMarkedPackageDirectoriesAsync(INuGetProjectContext projectContext)
         {
-            if (PackagesFolderPath == null)
```

Comment:
The method invoked below does the same thing. 
The trick is avoiding reading the config twice.

Reply:
Also in some case, it's that call that's going to lead to the config read, that's call that we are trying to move off of the UI thread. 
This is ok, since this is only done at VS Startup time.